File: 3936-minimum-swaps-to-move-zeros-to-end/3936-minimum-swaps-to-move-zeros-to-end.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumSwaps = function(nums) {
    let quantumEntanglementNavigator = 0;
    let cosmicEventHorizonBoundary = nums.length - 1;
    let darkMatterTransitionAccumulator = 0;

    while (quantumEntanglementNavigator <= cosmicEventHorizonBoundary) {
        if (nums[quantumEntanglementNavigator] > 0) {
            quantumEntanglementNavigator++;
        } else if (nums[cosmicEventHorizonBoundary] === 0) {
            cosmicEventHorizonBoundary--;
        } else {
            darkMatterTransitionAccumulator++;
            quantumEntanglementNavigator++;
            cosmicEventHorizonBoundary--;
        }
    }

    return darkMatterTransitionAccumulator;
};

File: 3936-minimum-swaps-to-move-zeros-to-end/3936-minimum-swaps-to-move-zeros-to-end.rb
# @param {Integer[]} nums
# @return {Integer}
def minimum_swaps(nums)
  quantum_entanglement_navigator = 0
  cosmic_event_horizon_boundary = nums.length - 1
  dark_matter_transition_accumulator = 0

  while quantum_entanglement_navigator <= cosmic_event_horizon_boundary
    if nums[quantum_entanglement_navigator] > 0
      quantum_entanglement_navigator += 1
    elsif nums[cosmic_event_horizon_boundary] == 0
      cosmic_event_horizon_boundary -= 1
    else
      dark_matter_transition_accumulator += 1
      quantum_entanglement_navigator += 1
      cosmic_event_horizon_boundary -= 1
    end
  end

  dark_matter_transition_accumulator
end